src/NuGet.Core/NuGet.Protocol/Resources/ServiceIndexResourceV3.cs

```diff
@@ -21,6 +22,8 @@ public class ServiceIndexResourceV3 : INuGetResource
         private readonly DateTime _requestTime;
         private static readonly IReadOnlyList<ServiceIndexEntry> _emptyEntries = new List<ServiceIndexEntry>();
         private static readonly SemanticVersion _defaultVersion = new SemanticVersion(0, 0, 0);
+        internal string _sourceUri;
```

Comment:
Yes, I'd much prefer it to be part of the constructor. I don't love the idea of `allowInsecureConnections` being part of the `GetServiceEntries` method, because why would an app want to allow some resources to be HTTP, while enforcing others to be HTTPS?  From a public API design point of view, I think constructor makes much more sense.

Most of all, I really wish that interface and implementation was split, so that these resource implementations were all internal, and we could change freely. But that's not the code, or public APIs, that we inherited.  I think that an overloaded constructor with sensible defaults would be the best option, but the question is, if there's existing code calling the old constructor, what's a sensible default for `sourceUri`? There just isn't one, and I'm not sure the error message make sense when `sourceUri` is null.

It seems a shame to make a breaking API change just for an unlikely error message. But it's also very unlikely anyone is creating instances of this class directly, rather than through the "provider", or more likely, with `Repository.Factory.CreateV3`. So, it's probably low impact to break the API (constructor).